spec.html

```diff
@@ -10205,15 +10205,15 @@ <h2>Syntax</h2>
       <p>ECMAScript differs from the Java programming language in the behaviour of Unicode escape sequences. In a Java program, if the Unicode escape sequence `\\u000A`, for example, occurs within a single-line comment, it is interpreted as a line terminator (Unicode code point U+000A is LINE FEED (LF)) and therefore the next code point is not part of the comment. Similarly, if the Unicode escape sequence `\\u000A` occurs within a string literal in a Java program, it is likewise interpreted as a line terminator, which is not allowed within a string literal&mdash;one must write `\\n` instead of `\\u000A` to cause a LINE FEED (LF) to be part of the String value of a string literal. In an ECMAScript program, a Unicode escape sequence occurring within a comment is never interpreted and therefore cannot contribute to termination of the comment. Similarly, a Unicode escape sequence occurring within a string literal in an ECMAScript program always contributes to the literal and is never interpreted as a line terminator or as a code point that might terminate the string literal.</p>
     </emu-note>
 
-    <emu-clause id="sec-codepointtoutf16codeunits" aoid="CodePointToUTF16CodeUnits" oldids="sec-utf16encoding">
-      <h1>Static Semantics: CodePointToUTF16CodeUnits ( _cp_ )</h1>
-      <p>The abstract operation CodePointToUTF16CodeUnits takes argument _cp_ (a numeric code point value). It performs the following steps when called:</p>
+    <emu-clause id="sec-codepointtostring" aoid="CodePointToString" oldids="sec-utf16encoding,sec-codepointtoutf16codeunits">
+      <h1>Static Semantics: CodePointToString ( _cp_ )</h1>
```

Comment:
I actually still kinda prefer calling this AO `CodePointToUTF16CodeUnits` since it's really clear why it's needed. @tc39/ecma262-editors?

Reply:
Agreed.

Reply:
I'm not sure that makes it more clear - the direct analog here is `String.fromCodePoint`.

Reply:
When I think about `fromCodePoint`, I am thinking about the fact that I put in a Number and get out a String. Here I want readers to think about the fact that they put in a code point and get out a sequence of code units of length potentially greater than 1.

Reply:
"a sequence of code units of length potentially greater than 1" is also called "a string".

Reply:
Yes, but calling it a string does not emphasize the fact that the length might be greater than 1. I think about `String.fromCodePoint` exactly like `String.fromCharCode`, even though the latter always returns string of length 1 (when given one argument).

Calling it `CodePointToUTF16CodeUnits` emphasizes that you can get multiple code units out, and I think that is worth emphasizing.

Reply:
I'm not convinced - I think "String" already implies that - but at the least, why not `CodePointToCodeUnits`?

Reply:
> I think "String" already implies that

... How so? We could just as easily have a "CharCodeToString" AO, and that would return strings of length exactly 1 despite having almost identical naming.

> why not `CodePointToCodeUnits` 

Because there are several different notions "code unit", so the name would ambiguous if you didn't specify.

Reply:
CharCodeToString would of course return a string of length 1, because a charcode represents a single code unit. A code point represents 1 or 2 code points, so it would return a string of length 1 or 2. I don't see why it needs emphasizing, or why "CodePointToUTF16CodeUnits" would make it clear that it _can_ return a string of length < 2 (to me, "units" is plural, and thus implies "more than one").

Reply:
The preamble could say that the operation returns a String consisting of 1 or 2 code units. Does that help?

Reply:
@jmdyck I don't think that really addresses it. I'm thinking about the reader understanding what the function does at each call site. I think using the term "string" within the preamble or the steps is fine since you have the explicit description of its behaviour right there.

Reply:
If we agree that unifying on Strings in favor of sequences of UTF16 code units is a good thing, and we did on the last editor call, then this AO should avoid `UTF16CodeUnits` in the name as that would be its own form of confusion.

The discussion seems to be stuck on whether the plural "Code Units" better primes the readers at callsites to think about the length of the result, which can be 1 or 2. I don't find the `CodePointToUTF16CodeUnits` name to be particularly clear on conveying that point.

There's also an argument that `CodePointToUTF16CodeUnits` is clearer on what the AO does than `ToString`. If that's the position, then it feels like you actually don't think unifying on Strings in favor of sequences of UTF16 code units is a good idea to begin with.


Maybe something like `UTF16EncodeCodePoint` instead?

Reply:
I don't like `UTF16EncodeCodePoint` as much as `CodePointToUTF16CodeUnits`, but I'd be OK with it; it's better than the other suggestions so far.